lib/omnibus/software.rb

```diff
@@ -483,13 +483,26 @@ def with_standard_compiler_flags(env = {}, opts = {})
       extra_linker_flags = {
         "LD_RUN_PATH" => "#{install_dir}/embedded/lib"
       }
-      # solaris linker can also use LD_OPTIONS, so we throw the kitchen sink against
-      # the linker, to find every way to make it use our rpath.
-      extra_linker_flags.merge!(
-        {
-          "LD_OPTIONS" => "-R#{install_dir}/embedded/lib"
-        }
-      ) if Ohai['platform'] == "solaris2"
+
+      if solaris2?
+        # in order to provide compatibility for earlier versions of libc on solaris 10,
+        # we need to specify a mapfile that restricts the version of system libraries
+        # used. See http://docs.oracle.com/cd/E23824_01/html/819-0690/chapter5-1.html
+        # for more information
+        # use the mapfile if it is configured, otherwise ignore it
+        unless Config.solaris_linker_mapfile.nil?
+         mapfile = " -M #{project.files_path}/#{Config.solaris_linker_mapfile}"
+        end 
```

Comment:
Per my other comment this can now change to:

```ruby
ld_options = "-R#{install_dir}/embedded/lib"
mapfile_path = File.expand_path(Config.solaris_linker_mapfile, Config.project_root)
ld_options  << " -M #{mapfile_path}" if File.exist?(mapfile_path)

extra_linker_flags.merge!(
  "LD_OPTIONS" => ld_options
)
```